en_US/durability/introduction.md

```diff
@@ -0,0 +1,93 @@
+when the client subscribes to a topic filter, EMQX marks topics matching the filter as "durable".
+In addition to routing the MQTT PUBLISH messages from such topics to the regular sessions, the broker will also save the messages on disc.
+
+Each durable MQTT message is stored on disc exactly once, regardless of the number of persistent sessions subscribing to the matching topic-filter, whether online or offline.
+This enables an efficient fan-out of messages to the persistent sessions.
+
+EMQX durable storage is partitioned into the following hierarchy:
+
+- **Shard**: messages published by different clients are stored in different shards.
+  Sharing is performed based on the client ID of the publisher.
+  Number of shards is determined by `session_persistence.storage.builtin.n_shards` configuration parameter during the first startup of the EMQX broker.
+
+- **Generation**:
+   each shard is additionally split into partitions called generations, each one covering a particular period of time.
+   New messages are written only into the _current_ generation, while the previous generations are only accessible for reading.
+
+   Different generations can organize the data in a different manner, according to the *storage layout* specification.
+   Currently only one layout is supported, that is optimized for the throughput of wildcard subscriptions covering very large number of topics, as well as the throughput of single-topic subscriptions.
+   In the future we will add other layouts to optimize for the different types of workloads, for example where low latency is more important than the high throughput.
+
+- **Stream**:
+   Finally, messages in each shard and generation are split into streams.
+   Streams can contain messages from multiple topics.
+   Various storage layouts can employ different strategies for mapping topics into streams.
+
+   Streams serve as units of message serialization in the EMQX broker.
+
+
+Persistent sessions read messages from the streams in batches.
+Size of the batch is configured by `session_persistence.max_batch_size` parameter.
+
+### Hardware requirements
+
+When session persistence is enabled, EMQX will save the metadata of persistent sessions, and any MQTT message sent to the persistent session on disc.
+It means that EMQX broker must be deployed on a server with sufficiently large storage capacity.
+To achieve the best throughput it's recommended to use SSD storage.
+
+Storage requirements can be estimated according to the following formulae:
+
+- Storage utilized by the messages is proportional to the rate of incoming messages multiplied by the value of configuration parameter `session_persistence.message_retention_period`.
+
+- Storage utilized by the persistent sessions is proportional to the number of sessions multiplied by the number of streams that the sessions subscribe to.
+
+- Number of streams is proportional to the number of shards.
+
+## Message retention
```

Comment:
Use the table to list all the configuration items, as well as the description of the configuration items

Reply:
Maybe adding a configuration code example with some explanations and what the effect will be after the configuration? Like with an MQTTX pub/sub example? Not sure whether it is necessary at this alpha stage. @wivwiv @ieQu1

Reply:
Another question is that don't we need to mention the built-in backend "RocksDB" at all in the document?

Reply:
> Another question is that don't we need to mention the built-in backend "RocksDB" at all in the document?

Good point. I think it's worth mentioning.

Reply:
@wivwiv Fixed in the latest commit.